lib/palindrome_check.rb

```diff
@@ -1,7 +1,28 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: ? O(n)
+# Space complexity: ? O(1)
```

Comment:
Please explain your thought process behind your answers for time and space complexity.